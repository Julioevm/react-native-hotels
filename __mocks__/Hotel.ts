export const Hotel = {
  id: '1',
  name: 'Hotel',
  stars: 4,
  userRating: 4.5,
  price: 100,
  currency: 'EUR',
  location: {
    latitude: 0,
    longitude: 0,
    city: 'Madrid',
    country: 'Spain',
    address: 'Calle de la Princesa, 1',
  },
  checkIn: {from: '2020-01-01', to: '2020-01-02'},
  checkOut: {from: '2020-01-01', to: '2020-01-02'},
  contact: {
    phoneNumber: '123456789',
    email: 'test@mail.com',
  },
  gallery: ['https://picsum.photos/200/300'],
};
