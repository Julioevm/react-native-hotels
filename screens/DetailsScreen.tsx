import {StyleSheet, Text, View} from 'react-native';
import React from 'react';
import {RouteProp} from '@react-navigation/native';
import {RootStackParamList} from '../types';

type Props = {
  route: RouteProp<RootStackParamList, 'Details'>;
};

export default function DetailsScreen({route}: Props) {
  const {hotel} = route.params;
  return (
    <View style={styles.container}>
      <Text>{hotel.name}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
