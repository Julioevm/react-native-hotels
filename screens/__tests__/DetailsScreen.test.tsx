import {render} from '@testing-library/react-native';
import {Hotels} from '../../__mocks__/Hotel';
import DetailsScreen from '../DetailsScreen';
import React from 'react';
jest.mock('../../components/common/Carrousel.tsx', () => 'Carrousel');

describe('DetailsScreen', () => {
  it('should render correctly', () => {
    const route = {
      params: {
        hotel: Hotels[0],
      },
    };
    const {toJSON} = render(<DetailsScreen route={route as any} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
