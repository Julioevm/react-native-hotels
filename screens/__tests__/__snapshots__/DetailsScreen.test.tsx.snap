// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 16,
      "paddingTop": 80,
    }
  }
>
  <Carrousel
    autoPlay={true}
    gallery={
      [
        "https://res.cloudinary.com/lastminute/image/upload/t_OSE_redes_item_view/v1499779963/Swindon_yjsrwz.jpg",
      ]
    }
    width={718}
  />
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginVertical": 8,
      }
    }
  >
    Park Plaza London Waterloo
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/star-small.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "width": 16,
            },
            {
              "tintColor": "orange",
            },
          ]
        }
      />
      <Image
        source={
          {
            "testUri": "../../../assets/star-small.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "width": 16,
            },
            {
              "tintColor": "orange",
            },
          ]
        }
      />
      <Image
        source={
          {
            "testUri": "../../../assets/star-small.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "width": 16,
            },
            {
              "tintColor": "orange",
            },
          ]
        }
      />
      <Image
        source={
          {
            "testUri": "../../../assets/star-small.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "width": 16,
            },
            {
              "tintColor": "orange",
            },
          ]
        }
      />
      <Image
        source={
          {
            "testUri": "../../../assets/star-small.png",
          }
        }
        style={
          [
            {
              "height": 16,
              "width": 16,
            },
            {
              "tintColor": "gray",
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text>
        User Rating
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 4,
              "height": 18,
              "justifyContent": "center",
              "marginLeft": 6,
              "width": 18,
            },
            {
              "backgroundColor": "hsl(108, 100%, 70%)",
            },
          ]
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            {
              "fontWeight": "bold",
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          9
        </Text>
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        120
         
        â‚¬
      </Text>
    </View>
  </View>
  <Text
    onPress={[Function]}
    style={
      {
        "textDecorationLine": "underline",
      }
    }
  >
    6 Hercules Road
    , 
    London
  </Text>
  <View
    style={
      {
        "backgroundColor": "lavenderblush",
        "borderRadius": 8,
        "marginTop": 8,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Contact:
    </Text>
    <Text>
      Phone: 
      +39 24322342
    </Text>
    <Text>
      Email: 
      park_plaza@gmail.com
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Check in:
    </Text>
    <Text>
      From 
      14:00
       to 
      20:00
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Check out: 
    </Text>
    <Text>
      From 
      07:00
       to 
      10:00
    </Text>
  </View>
  <View
    style={
      {
        "marginTop": 16,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "tomato",
              },
            ]
          }
        >
          Book Now
        </Text>
      </View>
    </View>
  </View>
</View>
`;
